packages/airnode-node/src/evm/fulfillments/index.ts

```diff
@@ -9,141 +7,147 @@ import {
-    const makeRequest = async () => {
-      const [logs, err, data] = await submitWithdrawal(contract, withdrawal, txOptions);
-      logger.logPending(logs, baseLogOptions);
-      if (err || !data) {
-        return { id: withdrawal.id, type: RequestType.Withdrawal, error: err };
-      }
-      return { id: withdrawal.id, type: RequestType.Withdrawal, data };
-    };
-
-    return {
-      nonce: withdrawal.nonce!,
-      makeRequest,
-    };
-  });
+  const submittedWithdrawals = submitRequests(
+    state,
+    verifiedWithdrawals,
+    RequestType.Withdrawal,
+    submitWithdrawal,
+    contract
+  );
 
   const allRequests = [...submittedApiCalls, ...submittedWithdrawals];
   // Sort by the nonce value increasingly
   allRequests.sort((a, b) => a.nonce - b.nonce);
 
-  const receipts: RequestReceipt[] = [];
+  const apiCalls: Request<ApiCall>[] = [];
+  const withdrawals: Request<Withdrawal>[] = [];
   // Perform the requests sequentially to in order to respect the nonce value
   for (const request of allRequests) {
-    receipts.push(await request.makeRequest());
+    const submittedRequest = await request.makeRequest();
+    if (submittedRequest.fulfillment?.hash) {
+      logger.info(
+        `Transaction:${submittedRequest.fulfillment.hash} submitted for Request:${submittedRequest.id}`,
+        baseLogOptions(state)
+      );
+    }
+
+    // TODO: Include RequestType in the actual Request object
+    if (request.type === RequestType.ApiCall) {
+      apiCalls.push(submittedRequest as Request<ApiCall>);
+    }
+    if (request.type === RequestType.Withdrawal) {
+      withdrawals.push(submittedRequest as Request<Withdrawal>);
+    }
   }
 
-  return receipts;
-};
+  return {
+    apiCalls,
+    withdrawals,
+  };
+}
 
 export async function submit(state: ProviderState<EVMProviderState>) {
   const requestsBySponsorAddress = grouping.groupRequestsBySponsorAddress(state.requests);
   const sponsorAddresses = Object.keys(requestsBySponsorAddress);
 
   const promises = sponsorAddresses.map((address) => submitSponsorRequestsSequentially(state, address));
-  const nestedReceipts = await Promise.all(promises);
-  return flatten(nestedReceipts);
-}
-
-export function applyFulfillments<T>(requests: Request<T>[], receipts: TransactionReceipt[]) {
-  return requests.reduce((acc, request) => {
-    const receipt = receipts.find((r) => r.id === request.id);
-    // If the request was not submitted or the transaction doesn't have a hash, leave it as is
-    if (!receipt || !receipt.data?.hash) {
-      return [...acc, request];
-    }
-
-    const updatedRequest = {
-      ...request,
-      fulfillment: { hash: receipt.data.hash },
-      status: RequestStatus.Submitted,
-    };
-    return [...acc, updatedRequest];
-  }, [] as Request<T>[]);
+  const nestedGroupRequests = await Promise.all(promises);
+  return nestedGroupRequests.reduce((merged, groupRequests) => ({
+    apiCalls: [...merged.apiCalls, ...groupRequests.apiCalls],
```

Comment:
This will result in a different order of requests that was there previously (before the requests were grouped by sponsor address)

I implemented this with `flatten` previously, which had the same problem, but now with a bit more insight of how the requests flow through Airnode, I think we should respect the order. However, I think the proper way to do this would be to save up indexes of the logs immediately when they are read by `getLogs` of the provider. WDYT?

Reply:
But why is the different order a problem? Isn't the order important only when dealing with nonces (which should be fine)? Anyway, that's probably out of the scope of this PR, isn't it? Or is the current ordering a bug?

Reply:
More of an idea of the ideal solution. Whenever the order is important you would have a way to sort them. This way, you have to rely on the ordering being preserved which is not nice. I will create an issue for this.

Reply:
https://api3dao.atlassian.net/browse/AN-452